src/libraries/RateLimiter.sol

```diff
@@ -165,30 +171,35 @@ abstract contract RateLimiter is IRateLimiter, IRateLimiterEvents {
      * @param currentCapacity The current capacity
      */
     function _calculateNewCurrentCapacity(
-        NormalizedAmount newLimit,
-        NormalizedAmount oldLimit,
-        NormalizedAmount currentCapacity
-    ) internal pure returns (NormalizedAmount newCurrentCapacity) {
-        NormalizedAmount difference;
-
-        if (oldLimit > newLimit) {
-            difference = oldLimit - newLimit;
-            newCurrentCapacity = currentCapacity > difference
-                ? currentCapacity - difference
-                : NormalizedAmount.wrap(0);
+        NormalizedAmount memory newLimit,
+        NormalizedAmount memory oldLimit,
+        NormalizedAmount memory currentCapacity
+    ) internal pure returns (NormalizedAmount memory newCurrentCapacity) {
+        NormalizedAmount memory difference;
+
+        if (oldLimit.gt(newLimit)) {
+            difference = oldLimit.sub(newLimit);
+            newCurrentCapacity = currentCapacity.gt(difference)
+                ? currentCapacity.sub(difference)
+                : NormalizedAmountLib.wrap(0, 0);
         } else {
-            difference = newLimit - oldLimit;
-            newCurrentCapacity = currentCapacity + difference;
+            difference = newLimit.sub(oldLimit);
+            newCurrentCapacity = currentCapacity.add(difference);
+        }
+
+        // TODO: change this to min
+        if (newCurrentCapacity.gt(newLimit)) {
+            revert CapacityCannotExceedLimit(newCurrentCapacity, newLimit);
```

Comment:
The TODO here says to change this to use `min` instead.
Why not keep the invariant check?

Reply:
yea i actually prefer the revert here so it propagates the error up the call stack. if the capacity exceeds the limit, something most likely went wrong